File: bug65_single_element_case_failure/bug65_single_element_case_failure/main.py
import base64, struct

def pairwise_xor(arr):
    result = []
    for i in range(len(arr) - 1):
        result.append(arr[i] ^ arr[i + 1])
    return result

def run():
    data = [0xABCD]
    pairs = pairwise_xor(data)
    if len(pairs) == 0:
        raise ValueError(f"pairwise_xor returned empty for single-element input {data}")
    acc = 0
    for i, v in enumerate(pairs):
        acc ^= v * (i + 1)
        acc = (acc * 0x9E3779B97F4A7C15) & 0xFFFFFFFFFFFFFFFF
    raw = struct.pack(">Q", acc)
    print(base64.b64encode(raw).decode())

run()

File: bug65_single_element_case_failure/bug65_single_element_case_failure/main.js
function main() {
    let data = [0xABCD];
    let pairs = [];
    for (let i = 0; i < data.length - 1; i++) {
        pairs.push(data[i] ^ data[i + 1]);
    }
    if (pairs.length === 0) {
        console.error(`ValueError: pairwise_xor returned empty for single-element input [${data[0]}]`);
        process.exit(1);
    }
    let acc = 0n;
    for (let i = 0; i < pairs.length; i++) {
        acc ^= BigInt(pairs[i]) * BigInt(i + 1);
        acc = (acc * 0x9E3779B97F4A7C15n) & 0xFFFFFFFFFFFFFFFFn;
    }
}
main();
